oligo_designer_toolsuite/oligo_specificity_filter/_filter_blastn.py

```diff
@@ -0,0 +1,289 @@
+                ]
+
+                if self.ligation_region == 0:
+                    blast_results_matches_probe_length = (
+                        blast_results_matches_probe_length[
+                            blast_results_matches_probe_length.alignment_length
+                            > min_alignment_length
+                        ]
+                    )
+                else:
+                    blast_results_matches_probe_length = (
+                        blast_results_matches_probe_length[
+                            (
+                                blast_results_matches_probe_length.alignment_length
+                                > min_alignment_length
+                            )
+                            | (
+                                (
+                                    blast_results_matches_probe_length.query_start
+                                    < blast_results_matches_probe_length.ligation_site
+                                    - (self.ligation_region - 1)
+                                )
+                                & (
+                                    blast_results_matches_probe_length.query_end
+                                    > (
+                                        blast_results_matches_probe_length.ligation_site
+                                        + self.ligation_region
+                                    )
+                                )
+                            )
+                        ]
+                    )
+
+                blast_results_matches_filtered.append(
+                    blast_results_matches_probe_length
+                )
+
+            blast_results_matches_filtered = pd.concat(blast_results_matches_filtered)
+
+            probes_with_match = blast_results_matches_filtered["query"].unique()
+            probes_wo_match = blast_results[
+                ~blast_results["query"].isin(probes_with_match)
+            ]
+
+            for gene_id in blast_results["query_gene_id"].unique():
+                probes_wo_match_gene = probes_wo_match[
+                    probes_wo_match.query_gene_id == gene_id
+                ]
+                probes_wo_match_gene = probes_wo_match_gene["query"].unique()
+
+                if len(probes_wo_match_gene) > 0:  # gene has to have at least one probe
+                    _write_output(self, probes_info, gene_id, probes_wo_match_gene)
+
+            return len(probes_wo_match["query"].unique())
+
+        # Parallelize filtering of probes
+        start_time = time.perf_counter()
+        with parallel_backend("loky"):
+            Parallel()(
+                delayed(_process_blast_results)(batch_id)
+                for batch_id in range(self.n_jobs)
+            )
+
+        finish_time = time.perf_counter()
+
+        self.logging.info(
+            f"Blast results processed in {finish_time-start_time} seconds"
+        )
+
+        _write_removed_genes(self)
```

Comment:
just remove all filtered probes from the database and keep genes with 0 probes, which can be written later from the database to a file